packages/front-end/pages/features/[fid].tsx

```diff
@@ -203,25 +205,79 @@ export default function FeaturePage() {
             >
               Show implementation
             </a>
-            {permissions.createFeatures && (
-              <DeleteButton
-                useIcon={false}
-                displayName="Feature"
+        {permissions.createFeatures && (<DeleteButton
+              useIcon={false}
+              displayName="Feature"
+              onClick={async () => {
+                await apiCall(`/feature/${data.feature.id}`, {
+                  method: "DELETE",
+                });
+                router.push("/features");
+              }}
+              className="dropdown-item"
+              text="Delete feature"
+            />) }
+            {(isArchived && permissions.createFeatures) ? (
```

Comment:
These two `<ConfirmButton>` components have a lot in common.  I would probably make it just render a single component and change some of the prop values depending on the `isArchived` state.  For example:

```tsx
<ConfirmButton
  modalHeader={isArchived ? "Unarchive Feature" : "Archive Feature"}
  ...
>
  <button...>{isArchived ? "Unarchive" : "Archive"} Feature</button>
</ConfirmButton>
```

Reply:
Changed in recent commit